# Homework 1 Problem 3

In $2d$ electron gas, using Born-von Karman boundary conditions:

\begin{equation}
\begin{aligned}
\psi (x,y + Y) = \psi(x,y) \\
\psi (x + X,y) = \psi(x,y)
\end{aligned}
\end{equation}

Therefore:

\begin{equation}
\psi_{\vec{k}}(\vec{r}) = \frac{e^{i\vec{k}\cdot\vec{r}}}{\sqrt{S}}
\end{equation}

Energy and momentum eigen values of electrons are related to wave vector:

\begin{equation}
\begin{aligned}
E_{\vec{k}} = \frac{\hbar^2 k^2}{2m}\\
\vec{p} = \hbar \vec{k}
\end{aligned}
\end{equation}

Recall boundary conditions. They imply:

\begin{equation}
\begin{aligned}
e^{ik_x X} = e^{ik_y Y} = 1
\end{aligned}
\end{equation}

Therefore:

\begin{equation}
\begin{aligned}
k_x = n_x \frac{2\pi}{X}\\
k_y = n_y \frac{2\pi}{Y}
\end{aligned}
\end{equation}

In $k-Space$, it's obvious that in every area of size $\frac{2\pi}{X}\frac{2\pi}{Y}$ there is only one grid point. That means that there are $\frac{S}{(2\pi)^2}$ grid points per unit area of $k-Space$. 

That further means that:

\begin{equation}
d^2 \vec{n} = \frac{S}{(2\pi)^2} d^2 \vec{k}
\end{equation}

Spin degeneracy is:

\begin{equation}
g_s = 2s+1 = 2
\end{equation}

Combining previous results, we can write equation for total number of electrons:

\begin{equation}
N = g_s \int \frac{d^2 \vec{k}}{(2\pi)^2} S \cdot f_{FD} (\vec{k})
\end{equation}

where Fermi-Dirac distribution is:

\begin{equation}
f_{FD} (\vec{k}) = \frac{1}{e^{\frac{E_{\vec{k}} - \mu}{kT}} + 1}
\end{equation}

Chemical potential at $T = 0K$ is called Fermi energy. To calculate it, we will use:

\begin{equation}
\lim_{T \to 0} f_{FD} (\vec{k}) = \lim_{T \to 0} \frac{1}{e^{\frac{E_{\vec{k}} - \mu}{kT}} + 1} = \theta(E_F - E_{\vec{k}})
\end{equation}

Using step function approximation of Fermi-Dirac distribution at low temperatures, we can easily calculate number of particles:

\begin{equation}
N = g_s \frac{S}{(2\pi)^2} \pi k_F^2
\end{equation}

Finally, as $n_S = \frac{N}{S}$, we obtain Fermi wave vector:

\begin{equation}
k_F = \sqrt{2\pi n_S}
\end{equation}

Knowing Fermi wave vector, we can easily calculate Fermi energy and Fermi velocity:

\begin{equation}
\begin{aligned}
E_F = \frac{\hbar^2 k_F^2}{2m} = \frac{\pi n_S \hbar^2}{m} \\
v_F = \frac{\hbar k_F}{m} = \frac{\hbar}{m} \sqrt{2\pi n_S}
\end{aligned}
\end{equation}

If we define $r_S$ by equation:

\begin{equation}
\frac{S}{N} = \frac{1}{n_S} = \pi r_S^2
\end{equation}

Then $n_S = \frac{1}{\pi r_S^2}$.

Bohr radius is:

\begin{equation}
a_0 = 0.529 \cdot 10^{-10}m
\end{equation}

In [1]:
a_0 = 0.529

k_F = 2 ** 0.5 / a_0

In [2]:
k_F #Unit is A^-1

2.6733715734841117

In [3]:
from scipy.constants import hbar, m_e, e

v_F = hbar / m_e * k_F * 10 ** 10

In [4]:
v_F

3094899.0720617673

In [5]:
e_F = hbar ** 2 * k_F ** 2 / (2 * m_e) * 10 ** 20
e_F/e

27.229620250262712

Therefore:

\begin{equation}
\begin{aligned}
k_F = \frac{1}{(r_S/a_0)}\cdot 2.67337 A^{-1}\\
v_F = \frac{1}{(r_S/a_0)}\cdot 3.00949 \cdot 10^{6} \frac{m}{s}\\
E_F = \frac{1}{(r_S/a_0)^2} \cdot 27.229 eV
\end{aligned}
\end{equation}

Degeneracy of electron gass is:

\begin{equation}
g(E) = \sum_{s, vec{k}} \delta(E_{\vec{k}} - E) \to g_s \int \frac{d^2 \vec{k}}{(2\pi)^2} S \cdot \delta(E_k - E)
\end{equation}

Changing variables $d^2 \vec{k} \to k dk d\phi$ and integrating over polar angle, we get:

\begin{equation}
g(E) =  \frac{g_s S}{2\pi} \int_0^{\infty} k dk \delta(E_k - E)
\end{equation}

Using $\delta(f(x)) = \frac{\delta(x-x_0)}{|f'(x_0)|}$, we get:

\begin{equation}
g(E) = \frac{g_s S}{2\pi} \frac{m}{\hbar^2} \int_0^{\infty} dk \delta(k - \frac{1}{\hbar}\sqrt{2mE})
\end{equation}

Finally:

\begin{equation}
g(E) = \frac{g_s m S}{2\pi \hbar^2}
\end{equation}

# Homework 1 Problem 4

Recall some expressions from previous problem.

In [6]:
from scipy.constants import k, e, epsilon_0, m_e, hbar, pi
from sympy import pprint
import numpy as np

def fermi_temperature(concentration):
    return fermi_energy(concentration) / k

def fermi_energy(concentration):
    return pi * concentration * hbar ** 2 / m_e

In [7]:
n_s = [10 ** 19, 10 ** 14]

In [8]:
for n in n_s:
    print("concentration = ", str(n), r"$m^{-2}$")
    print("Fermi Temperature = ", fermi_temperature(n), "K")

concentration =  10000000000000000000 $m^{-2}$
Fermi Temperature =  27779.860455981565 K
concentration =  100000000000000 $m^{-2}$
Fermi Temperature =  0.27779860455981564 K


Estimate of electrostatic energy is:

\begin{equation}
E_e = \frac{e^2}{4 \pi \epsilon_0 r_s}
\end{equation}

In [9]:
e_e = e ** 2 / (4 * pi * epsilon_0 * a_0)
e_e/e * 10 ** 10

27.220501610492526

In terms of $r_s/a_0$, we have:

\begin{equation}
E_e = \frac{1}{r_s/a_0} \cdot 27.2205 eV
\end{equation}

In [10]:
e_e/e_F * 10 ** 10

0.9996651205677355

We conclude that:

\begin{equation}
\frac{E_e}{E_F} \approx \frac{r_S}{a_0}
\end{equation}

In [11]:
def r_s(concentration):
    return 1 / np.sqrt(pi * concentration)

def electrostatic_energy(concentration):
    r = r_s(concentration)
    return e ** 2 / (4 * pi * epsilon_0 * r)

In [12]:
for n in n_s:
    print("concentration = ", str(n), r"$m^{-2}$")
    print("Fermi Energy = ", fermi_energy(n)/e, "eV")
    print("Electrostatic Energy = ", electrostatic_energy(n)/e, "eV")
    print("Ee/EF = ", electrostatic_energy(n)/fermi_energy(n))

concentration =  10000000000000000000 $m^{-2}$
Fermi Energy =  2.393882342709907 eV
Electrostatic Energy =  8.070988571692807 eV
Ee/EF =  3.371505954029611
concentration =  100000000000000 $m^{-2}$
Fermi Energy =  2.3938823427099072e-05 eV
Electrostatic Energy =  0.025522706855738462 eV
Ee/EF =  1066.163795955252


In case of smaller concentrations, electrostatic energy is more important then in case of smaller concentrations.